src/display/api.js

```diff
@@ -1742,7 +1751,10 @@ var WorkerTransport = (function WorkerTransportClosure() {
             var size = width * height;
             var rgbaLength = size * 4;
             var buf = new Uint8Array(size * components);
-            var tmpCanvas = createScratchCanvas(width, height);
+            // make sure that this code is not executing in node.js.
+            assert(typeof document !== 'undefined');
+            var canvasFactory = new DOMCanvasFactory();
```

Comment:
We probable need to make it `var tmpCanvas = document.createElement('canvas'); tmpCanvas.width =...`